pkg/scheduler/framework/plugins/interpodaffinity/scoring_test.go

```diff
@@ -773,31 +781,36 @@ func TestPreferredAffinity(t *testing.T) {
 			state := framework.NewCycleState()
 			p := plugintesting.SetupPluginWithInformers(ctx, t, New, &config.InterPodAffinityArgs{HardPodAffinityWeight: 1, IgnorePreferredTermsOfExistingPods: test.ignorePreferredTermsOfExistingPods}, cache.NewSnapshot(test.pods, test.nodes), namespaces)
 			status := p.(framework.PreScorePlugin).PreScore(ctx, state, test.pod, test.nodes)
+
 			if !status.IsSuccess() {
-				if !strings.Contains(status.Message(), test.wantStatus.Message()) {
-					t.Errorf("unexpected error: %v", status)
+				if status.Code() != test.wantStatus.Code() {
+					t.Errorf("InterPodAffinity#PreScore() returned unexpected status.Code got: %v, want: %v", status.Code(), test.wantStatus.Code())
 				}
-			} else {
-				var gotList framework.NodeScoreList
-				for _, n := range test.nodes {
-					nodeName := n.ObjectMeta.Name
-					score, status := p.(framework.ScorePlugin).Score(ctx, state, test.pod, nodeName)
-					if !status.IsSuccess() {
-						t.Errorf("unexpected error: %v", status)
-					}
-					gotList = append(gotList, framework.NodeScore{Name: nodeName, Score: score})
+
+				if !strings.Contains(status.Message(), test.wantStatus.Message()) {
+					t.Errorf("InterPodAffinity#PreScore() returned unexpected status.Message got: %v, want: %v", status.Message(), test.wantStatus.Message())
 				}
+				return
+			}
 
-				status = p.(framework.ScorePlugin).ScoreExtensions().NormalizeScore(ctx, state, test.pod, gotList)
+			var gotList framework.NodeScoreList
+			for _, n := range test.nodes {
+				nodeName := n.ObjectMeta.Name
+				score, status := p.(framework.ScorePlugin).Score(ctx, state, test.pod, nodeName)
 				if !status.IsSuccess() {
 					t.Errorf("unexpected error: %v", status)
 				}
+				gotList = append(gotList, framework.NodeScore{Name: nodeName, Score: score})
+			}
 
-				if !reflect.DeepEqual(test.expectedList, gotList) {
-					t.Errorf("expected:\n\t%+v,\ngot:\n\t%+v", test.expectedList, gotList)
-				}
+			status = p.(framework.ScorePlugin).ScoreExtensions().NormalizeScore(ctx, state, test.pod, gotList)
+			if !status.IsSuccess() {
+				t.Errorf("unexpected error: %v", status)
 			}
 
+			if !reflect.DeepEqual(test.expectedList, gotList) {
```

Comment:
Sorry, can we use cmp.Diff() instead since we're refactoring. Another suggestion is we can make refactoring another commit for better reviewing since it's independent. What I learned from the community, next time I think :).

Reply:
@kerthcet Sure! I have fixed it.